cmd/kanikoExecute.go

```diff
@@ -134,117 +135,207 @@ func runKanikoExecute(config *kanikoExecuteOptions, telemetryData *telemetry.Cus
+			if err = runKaniko(file, buildOpts, config.ReadImageDigest, execRunner, fileUtils, commonPipelineEnvironment); err != nil {
+				return fmt.Errorf("failed to build image '%v' using '%v': %w", image, file, err)
 			}
+			commonPipelineEnvironment.container.imageNames = append(commonPipelineEnvironment.container.imageNames, image)
+			commonPipelineEnvironment.container.imageNameTags = append(commonPipelineEnvironment.container.imageNameTags, containerImageNameAndTag)
+		}
 
-			commonPipelineEnvironment.container.registryURL = config.ContainerRegistryURL
+		// for compatibility reasons also fill single imageNameTag field with "root" image in commonPipelineEnvironment
+		// only consider if it has been built
+		// ToDo: reconsider and possibly remove at a later point
+		if len(imageListWithFilePath[config.ContainerImageName]) > 0 {
+			containerImageNameAndTag := fmt.Sprintf("%v:%v", config.ContainerImageName, containerImageTag)
+			commonPipelineEnvironment.container.imageNameTag = containerImageNameAndTag
+		}
+		if config.CreateBOM {
+			// Syft for multi image, generates bom-docker-(1/2/3).xml
+			return syft.GenerateSBOM(config.SyftDownloadURL, "/kaniko/.docker", execRunner, fileUtils, httpClient, commonPipelineEnvironment.container.registryURL, commonPipelineEnvironment.container.imageNameTags)
+		}
+		return nil
 
-			// Docker image tags don't allow plus signs in tags, thus replacing with dash
-			containerImageTag := strings.ReplaceAll(config.ContainerImageTag, "+", "-")
+	case config.MultipleImages != nil:
+		log.Entry().Debugf("MultipleImages build activated")
+		parsedMultipleImages, err := parseMultipleImages(config.MultipleImages)
+		if err != nil {
+			log.SetErrorCategory(log.ErrorConfiguration)
+			return errors.Wrap(err, "failed to parse MultipleImages")
+		}
 
-			if config.ContainerMultiImageBuild {
-				log.Entry().Debugf("Multi-image build activated for image name '%v'", config.ContainerImageName)
-				imageListWithFilePath, err := docker.ImageListWithFilePath(config.ContainerImageName, config.ContainerMultiImageBuildExcludes, config.ContainerMultiImageBuildTrimDir, fileUtils)
+		for _, entry := range parsedMultipleImages {
+			switch {
+			case entry.ContextSubPath == "":
+				return fmt.Errorf("empty ContextSubPath")
+			case entry.ContainerImageName != "":
+				containerRegistry, err := docker.ContainerRegistryFromURL(config.ContainerRegistryURL)
 				if err != nil {
-					return fmt.Errorf("failed to identify image list for multi image build: %w", err)
+					log.SetErrorCategory(log.ErrorConfiguration)
+					return errors.Wrapf(err, "failed to read registry url %v", config.ContainerRegistryURL)
 				}
-				if len(imageListWithFilePath) == 0 {
-					return fmt.Errorf("no docker files to process, please check exclude list")
+
+				if entry.ContainerImageTag == "" {
+					entry.ContainerImageTag = config.ContainerImageTag
 				}
```

Comment:
For not GPP cases the `config.containerImageTag` might be missing (empty), should we check if the tag is filled, otherwise return an error?